src/ufo.ts

```diff
@@ -0,0 +1,99 @@
+import { hasProtocol, parseURLNative, ParamsObject } from './parse'
+import { withoutLeadingSlash, withLeadingSlash, withTrailingSlash } from './utils'
+import { encodeSearchParam } from './encoding'
+
+export class UFO implements URL {
+   params: ParamsObject = {}
+   hash: string;
+   hostname: string;
+   password: string;
+   pathname: string;
+   port: string;
+   protocol: string;
+   username: string;
+
+   constructor (input: string = '') {
+     if (typeof input !== 'string') {
+       throw new TypeError(`URL input should be string recived ${typeof input} (${input})`)
```

Comment:
recived -> received

Reply:
We may want to JSON.stringify() the value :)

Reply:
If input is not serializable, it might make untraceable error